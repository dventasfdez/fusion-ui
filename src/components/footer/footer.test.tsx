import React from "react";
import Footer, { FooterBottom, FooterBrand, FooterHeader, FooterLogo } from "./footer";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import { fireEvent, waitFor } from "@testing-library/dom";

const footerExample = (props?: any) => {
  return (
    <Footer>
      <FooterHeader>
        <div>
          <a href="#">Link</a>
          <a href="#">Link</a>
          <a href="#">Link</a>
          <a href="#">Link</a>
        </div>
      </FooterHeader>
      <FooterBottom>
        <FooterLogo data-testid="footer-logo-test">
          <img
            style={{ objectFit: "contain" }}
            alt="footer-logo"
            src="https://www-prd-amz930-com.azureedge.net/es-es/-/media/project/adeccogroup/horizontal-the-adecco-group-brand-mark-land-rgb.png?h=475&w=1385&modified=00010101000000&hash=950A28C29AD61BBB9CB374DA671A06D1"
          />
        </FooterLogo>
        <a href="#">Link</a>
        <a href="#">Link</a>
        <a href="#">Link</a>
        <a href="#">Link</a>
        <FooterBrand data-testid="footer-brand-test">
          <p>@brand2022</p>
        </FooterBrand>
      </FooterBottom>
    </Footer>
  );
};

it("Footer should render", () => {
  const component = renderer.create(footerExample());
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("Display footer with logo", () => {
  const { getByTestId } = render(footerExample());
  const logo = getByTestId("footer-logo-test");
  expect(logo).toBeDefined();
});

it("Display footer with brand", () => {
  const { getByTestId } = render(footerExample());
  const brand = getByTestId("footer-brand-test");
  expect(brand).toBeDefined();
});

it("Display footer without children", () => {
  // @ts-expect-error No children
  const _ = render(<Footer />);
  expect(_).toBeDefined();
});
