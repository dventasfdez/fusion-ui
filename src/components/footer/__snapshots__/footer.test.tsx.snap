// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render 1`] = `
<div
  className="footer-container "
  data-testid="footer-test"
>
  <div
    className="footer-header"
  >
    <div
      className="footer-useful-links "
      data-testid="footer-useful-links-test"
    >
      <a
        href="#"
      >
        Useful Link 1
      </a>
      <a
        href="#"
      >
        Useful Link 2
      </a>
    </div>
    <div
      className="footer-social-media "
      data-testid="footer-social-media-links-test"
    >
      <a
        href="#"
      >
        SocialMedia Link 1
      </a>
      <a
        href="#"
      >
        SocialMedia Link 2
      </a>
    </div>
  </div>
  <hr
    className="mt4 mb4"
  />
  <div
    className="footer-bottom"
  >
    <img
      alt=""
      className="footer-logo  "
      data-testid="footer-logo-test"
      src="https://www-prd-amz930-com.azureedge.net/es-es/-/media/project/steponegroup/horizontal-the-stepone-group-brand-mark-land-rgb.png?h=475&w=1385&modified=00010101000000&hash=950A28C29AD61BBB9CB374DA671A06D1"
    />
    <div
      className="footer-policies-links "
      data-testid="footer-policies-links-test"
    >
      <a
        href="#"
      >
        Policy Link 1
      </a>
      <a
        href="#"
      >
        Policy Link 2
      </a>
    </div>
    <span
      className="footer-brand"
      data-testid="footer-brand-test"
    >
      stepone
    </span>
  </div>
</div>
`;

exports[`Footer without brand children should render 1`] = `
<div
  className="footer-container "
  data-testid="footer-test"
>
  <div
    className="footer-header"
  >
    <div
      className="footer-useful-links "
      data-testid="footer-useful-links-test"
    >
      <a
        href="#"
      >
        Useful Link 1
      </a>
      <a
        href="#"
      >
        Useful Link 2
      </a>
    </div>
    <div
      className="footer-social-media "
      data-testid="footer-social-media-links-test"
    >
      <a
        href="#"
      >
        SocialMedia Link 1
      </a>
      <a
        href="#"
      >
        SocialMedia Link 2
      </a>
    </div>
  </div>
  <hr
    className="mt4 mb4"
  />
  <div
    className="footer-bottom"
  >
    <img
      alt=""
      className="footer-logo  "
      data-testid="footer-logo-test"
      src="https://www-prd-amz930-com.azureedge.net/es-es/-/media/project/steponegroup/horizontal-the-stepone-group-brand-mark-land-rgb.png?h=475&w=1385&modified=00010101000000&hash=950A28C29AD61BBB9CB374DA671A06D1"
    />
    <div
      className="footer-policies-links "
      data-testid="footer-policies-links-test"
    >
      <a
        href="#"
      >
        Policy Link 1
      </a>
      <a
        href="#"
      >
        Policy Link 2
      </a>
    </div>
  </div>
</div>
`;

exports[`Footer without logo children should render 1`] = `
<div
  className="footer-container "
  data-testid="footer-test"
>
  <div
    className="footer-header"
  >
    <div
      className="footer-useful-links "
      data-testid="footer-useful-links-test"
    >
      <a
        href="#"
      >
        Useful Link 1
      </a>
      <a
        href="#"
      >
        Useful Link 2
      </a>
    </div>
    <div
      className="footer-social-media "
      data-testid="footer-social-media-links-test"
    >
      <a
        href="#"
      >
        SocialMedia Link 1
      </a>
      <a
        href="#"
      >
        SocialMedia Link 2
      </a>
    </div>
  </div>
  <hr
    className="mt4 mb4"
  />
  <div
    className="footer-bottom"
  >
    <div
      className="footer-policies-links "
      data-testid="footer-policies-links-test"
    >
      <a
        href="#"
      >
        Policy Link 1
      </a>
      <a
        href="#"
      >
        Policy Link 2
      </a>
    </div>
    <span
      className="footer-brand"
      data-testid="footer-brand-test"
    >
      stepone
    </span>
  </div>
</div>
`;

exports[`Footer without policies should render 1`] = `
<div
  className="footer-container "
  data-testid="footer-test"
>
  <div
    className="footer-header"
  >
    <div
      className="footer-useful-links "
      data-testid="footer-useful-links-test"
    >
      <a
        href="#"
      >
        Useful Link 1
      </a>
      <a
        href="#"
      >
        Useful Link 2
      </a>
    </div>
    <div
      className="footer-social-media "
      data-testid="footer-social-media-links-test"
    >
      <a
        href="#"
      >
        SocialMedia Link 1
      </a>
      <a
        href="#"
      >
        SocialMedia Link 2
      </a>
    </div>
  </div>
  <hr
    className="mt4 mb4"
  />
  <div
    className="footer-bottom"
  >
    <img
      alt=""
      className="footer-logo  "
      data-testid="footer-logo-test"
      src="https://www-prd-amz930-com.azureedge.net/es-es/-/media/project/steponegroup/horizontal-the-stepone-group-brand-mark-land-rgb.png?h=475&w=1385&modified=00010101000000&hash=950A28C29AD61BBB9CB374DA671A06D1"
    />
    <span
      className="footer-brand"
      data-testid="footer-brand-test"
    >
      stepone
    </span>
  </div>
</div>
`;

exports[`Footer without social media links should render 1`] = `
<div
  className="footer-container "
  data-testid="footer-test"
>
  <div
    className="footer-header"
  >
    <div
      className="footer-useful-links "
      data-testid="footer-useful-links-test"
    >
      <a
        href="#"
      >
        Useful Link 1
      </a>
      <a
        href="#"
      >
        Useful Link 2
      </a>
    </div>
  </div>
  <hr
    className="mt4 mb4"
  />
  <div
    className="footer-bottom"
  >
    <img
      alt=""
      className="footer-logo  "
      data-testid="footer-logo-test"
      src="https://www-prd-amz930-com.azureedge.net/es-es/-/media/project/steponegroup/horizontal-the-stepone-group-brand-mark-land-rgb.png?h=475&w=1385&modified=00010101000000&hash=950A28C29AD61BBB9CB374DA671A06D1"
    />
    <div
      className="footer-policies-links "
      data-testid="footer-policies-links-test"
    >
      <a
        href="#"
      >
        Policy Link 1
      </a>
      <a
        href="#"
      >
        Policy Link 2
      </a>
    </div>
    <span
      className="footer-brand"
      data-testid="footer-brand-test"
    >
      stepone
    </span>
  </div>
</div>
`;

exports[`Footer without test ids and class name in brand should render 1`] = `
<div
  className="footer-container "
>
  <div
    className="footer-header"
  >
    <div
      className="footer-useful-links "
    >
      <a
        href="#"
      >
        Useful Link 1
      </a>
      <a
        href="#"
      >
        Useful Link 2
      </a>
    </div>
    <div
      className="footer-social-media "
    >
      <a
        href="#"
      >
        SocialMedia Link 1
      </a>
      <a
        href="#"
      >
        SocialMedia Link 2
      </a>
    </div>
  </div>
  <hr
    className="mt4 mb4"
  />
  <div
    className="footer-bottom"
  >
    <img
      alt=""
      className="footer-logo test "
      src="https://www-prd-amz930-com.azureedge.net/es-es/-/media/project/steponegroup/horizontal-the-stepone-group-brand-mark-land-rgb.png?h=475&w=1385&modified=00010101000000&hash=950A28C29AD61BBB9CB374DA671A06D1"
    />
    <div
      className="footer-policies-links "
    >
      <a
        href="#"
      >
        Policy Link 1
      </a>
      <a
        href="#"
      >
        Policy Link 2
      </a>
    </div>
    <p
      className="footer-brand  "
    >
      stepone
    </p>
  </div>
</div>
`;

exports[`Footer without test ids should render 1`] = `
<div
  className="footer-container "
>
  <div
    className="footer-header"
  >
    <div
      className="footer-useful-links "
    >
      <a
        href="#"
      >
        Useful Link 1
      </a>
      <a
        href="#"
      >
        Useful Link 2
      </a>
    </div>
    <div
      className="footer-social-media "
    >
      <a
        href="#"
      >
        SocialMedia Link 1
      </a>
      <a
        href="#"
      >
        SocialMedia Link 2
      </a>
    </div>
  </div>
  <hr
    className="mt4 mb4"
  />
  <div
    className="footer-bottom"
  >
    <img
      alt=""
      className="footer-logo test "
      src="https://www-prd-amz930-com.azureedge.net/es-es/-/media/project/steponegroup/horizontal-the-stepone-group-brand-mark-land-rgb.png?h=475&w=1385&modified=00010101000000&hash=950A28C29AD61BBB9CB374DA671A06D1"
    />
    <div
      className="footer-policies-links "
    >
      <a
        href="#"
      >
        Policy Link 1
      </a>
      <a
        href="#"
      >
        Policy Link 2
      </a>
    </div>
    <p
      className="footer-brand test "
    >
      stepone
    </p>
  </div>
</div>
`;

exports[`Footer without useful links should render 1`] = `
<div
  className="footer-container "
  data-testid="footer-test"
>
  <div
    className="footer-header"
  >
    <div
      className="footer-social-media "
      data-testid="footer-social-media-links-test"
    >
      <a
        href="#"
      >
        SocialMedia Link 1
      </a>
      <a
        href="#"
      >
        SocialMedia Link 2
      </a>
    </div>
  </div>
  <hr
    className="mt4 mb4"
  />
  <div
    className="footer-bottom"
  >
    <img
      alt=""
      className="footer-logo  "
      data-testid="footer-logo-test"
      src="https://www-prd-amz930-com.azureedge.net/es-es/-/media/project/steponegroup/horizontal-the-stepone-group-brand-mark-land-rgb.png?h=475&w=1385&modified=00010101000000&hash=950A28C29AD61BBB9CB374DA671A06D1"
    />
    <div
      className="footer-policies-links "
      data-testid="footer-policies-links-test"
    >
      <a
        href="#"
      >
        Policy Link 1
      </a>
      <a
        href="#"
      >
        Policy Link 2
      </a>
    </div>
    <span
      className="footer-brand"
      data-testid="footer-brand-test"
    >
      stepone
    </span>
  </div>
</div>
`;
